docs/sections/user_guide/uw_yaml/forecast_yaml.rst

```diff
@@ -0,0 +1,232 @@
+        write_dopost: .false.
+    namelist:
+      base_file: /path/to/base/input.nml
+      update_values:
+        fv_core_nml:
+          k_split: 2
+          n_split: 6
+    nems.configure:
+      template_file: /path/to/template/nems.configure
+    run_dir: /path/to/forecast/run/directory/{yyyymmddhh}
+    static:
+      fv3.exe: /path/to/executable/ufs_model
+      INPUT/grid_spec.nc: /path/to/grid_spec.nc
+      ...
+      data_table: /path/to/data_table
+      eta_micro_lookup.data: /path/to/noahmptable.dat
+      noahmptable.tbl: /path/to/noahmptable.tbl
+
+.. _updating_values:
+
+Updating Values
+^^^^^^^^^^^^^^^
+
+Many of the sections describe configuration files needed by the UFS Weather Model, i.e. ``namelist:``, ``fd_ufs:``, ``model_configure:``. The ``base_file:`` sub-entry is required to initially stage the file; it can then be modified via an ``update_values:`` block.
+
+To ensure the correct values are updated, all keys that act as section headers above the entry to be updated need to be provided in the order in which they appear in the base file. Multiple entries within a block may be updated and they need not follow the same order as those in the base file. For example, the base file named ``people.yaml`` may contain:
+
+.. code-block:: yaml
+
+   person:
+     age: 19
+     address:
+       city: Boston
+       number: 12
+       state: MA
+       street: Acorn St
+     name: Jane
+
+Then the entries under the YAML section would edit this base file with the entries:
+
+.. code-block:: yaml
+
+   base_file: people.yaml
+   update_values:
+     person:
+       address:
+         street: Main St
+         number: 99
+
+Rendering Template Files
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires a path to a template file in the ``template_file:`` entry. There is no option to add values in the YAML. Instead, the driver is programmed to enter necessary values for the template.
+
+UW YAML Keys
+^^^^^^^^^^^^
+
+``cycle-dependent:``
+""""""""""""""""""""
+
+This block contains a set of files to stage in the run directory: file names as they appear in the run directory are keys and their source paths are the values. Source paths can be provided as a single string path, or a list of paths to be staged in a common directory under their original names.
+
+  .. warning:: The beta version does not support adding cycle information to the content of the files, and this information must be hard-coded in the YAML file.
+
+``diag_table:``
+"""""""""""""""
+
+In ``uwtools``, the ``diag_table`` is treated as a template so that the date and time information in the header may be filled in appropriately. The ``template_file:`` is the path to the input Jinja2 template. Date information is provided in the command line or API interfaces.
+
+The diag_table is described :weather-model-io:`here<diag-table-file>`.
+
+``domain:``
+"""""""""""
+
+A switch to differentiate between a global or regional configuration. Accepted values are ``global`` and ``regional``.
```

Comment:
I added `domain` to the JSON schema as a string with an enum. Is there another step I missed?

Reply:
Comparing the `FV3Forecast.jsonschema` in this PR to the YAML block [here](https://uwtools--389.org.readthedocs.build/en/389/sections/user_guide/uw_yaml/forecast_yaml.html#the-forecast-section), keys I do not see in the schema are: `diag_table`, `template_file`, `fd_ufs`, `nems.configure` (which I'd like to see as `nems_configure`).

I'd think that any key that we know in advance should be defined in the schema. Obviously we can't do keys that correspond to, say, namelist variables that we can't know in advance (because a model developer could add new ones, and will not be in a position to edit JSON Schema to get `uw` to accept those new variables).

I also think that we should define the structure of `field_table.update_values` further. If I understood the docs I read, anyway, we know that the structure has to contain `longname`, `units`, etc. keys -- even though we don't know the key name, which is fine because we can define _that_ with a regex. But the sub-structure is fixed, right? It will be very helpful for our users if we can get them as close as possible to correctness via schema checking.

I also don't see `additionalProperties: false` set anywhere here, so users can confuse themselves with misspellings, adding things at the wrong level, etc. I think we should be setting `additionalProperties: false` for every `object` where the set of acceptable keys is fixed, and we should define those keys.

So, there's that, in terms of completeness. I think this schema has been languishing for a long time while we improved `rocoto.jsonschema`. It's probably time to renovate this one using some best practices we've learned.

Apologies if I'm missing something about why this schema can't be locked down more.

One last thing: I've been adding [schema tests](https://uwtools--389.org.readthedocs.build/en/389/sections/user_guide/uw_yaml/forecast_yaml.html#the-forecast-section) for the Rocoto schema as a way to test very granular schema correctness without having to put together big YAML blocks. We may want to do the same for the FV3 schema.

Possibly all of this should be a separate ticket, or multiple tickets. Assuming I'm not wrong about it being desired/needed...

Reply:
It would be lovely to get that in a 2nd go, if you're okay with me capturing that in the Jira backlog.

Reply:
Good plan.